code/harness/DiskContents.cpp

```diff
@@ -106,22 +91,20 @@ ofstream& operator<< (ofstream& os, fileAttributes& a) {
     os << "Type   : " << (a.dir_attr)->d_type << endl;
   }
   // print stat_attr
-  if (a.stat_attr != NULL) {
-    os << "---File Stat Atrributes---" << endl;
-    os << "Inode     : " << (a.stat_attr)->st_ino << endl;
-    os << "TotalSize : " << (a.stat_attr)->st_size << endl;
-    os << "BlockSize : " << (a.stat_attr)->st_blksize << endl;
-    os << "#Blocks   : " << (a.stat_attr)->st_blocks << endl;
-    os << "#HardLinks: " << (a.stat_attr)->st_nlink << endl;
-    os << "Mode      : " << (a.stat_attr)->st_mode << endl;
-    os << "User ID   : " << (a.stat_attr)->st_uid << endl;
-    os << "Group ID  : " << (a.stat_attr)->st_gid << endl;
-    os << "Device ID : " << (a.stat_attr)->st_rdev << endl;
-    os << "RootDev ID: " << (a.stat_attr)->st_dev << endl;
-  }
+  os << "---File Stat Atrributes---" << endl;
+  os << "Inode     : " << (a.stat_attr).st_ino << endl;
+  os << "TotalSize : " << (a.stat_attr).st_size << endl;
+  os << "BlockSize : " << (a.stat_attr).st_blksize << endl;
+  os << "#Blocks   : " << (a.stat_attr).st_blocks << endl;
+  os << "#HardLinks: " << (a.stat_attr).st_nlink << endl;
+  os << "Mode      : " << (a.stat_attr).st_mode << endl;
+  os << "User ID   : " << (a.stat_attr).st_uid << endl;
+  os << "Group ID  : " << (a.stat_attr).st_gid << endl;
+  os << "Device ID : " << (a.stat_attr).st_rdev << endl;
+  os << "RootDev ID: " << (a.stat_attr).st_dev << endl;
 }
 
-DiskContents::DiskContents(char* path, const char* type) {
+DiskContents::DiskContents(const char* path, const char* type) {
   disk_path = (char *) malloc(sizeof(char)*30);
```

Comment:
nit: I would make these all `string` type in the header file (by declaring them `std::string` since we try not to have `using X` blocks in the header files) and then just use `string(path)` etc. to assign to them, that way you don't have to worry about memory management for them.